### PR TITLE
mypy: disallow untyped defs

### PR DESCRIPTION
**Pull request by Akuli**
When working on #20, I noticed that mypy wasn't creating an error when I expected it to create an error. The problem was that the `__init__` method of that class was like this:

```
    def __init__(self, app):
        self.app = app
        self.page = Page(self.app)
```

This means that `app` has type `Any` (which means "mypy, please don't complain about the type of this object"). Then `self.app` also has type `Any`, and you can do pretty much anything to `self.app` without getting a mypy error. A similar thing can happen every time when you have a `def` without type hints.

To fix this, I changed the mypy config to require every `def` to have type hints, so that nothing would weirdly default to `Any`. I then fixed the resulting mypy errors, and I think one of them was a bug that can crash pyhstr.

### CODE REVIEW

**Review by adder46 — APPROVED** [review]

File: pyhstr/user_interface.py [hunk 1]
```diff
@@ -1,17 +1,18 @@
 import curses
 import sys
 from typing import Dict, List, Optional, Union
+from pyhstr.application import App
```
Comment (adder46) [inline pyhstr/user_interface.py:4]: This line causes pylint to report a cyclic import problem.

File: pyhstr/application.py [hunk 2]
```diff
@@ -57,7 +57,7 @@
 
 
 class App:
-    def __init__(self, stdscr):
+    def __init__(self, stdscr: curses._CursesWindow):
```
Comment (adder46) [inline pyhstr/application.py:60]: pylint has a problem with this too, although I'm not sure how severe the actual issue is.

```
W0212: Access to a protected member _CursesWindow of a client class (protected-access)
```

Comment (Akuli) [reply pyhstr/application.py:60]: Seems like `curses._CursesWindow` is the right thing to do, although it is a bit weird: https://github.com/python/typeshed/issues/3116

**Comment by Akuli** [comment]
I will make a new PR that fixes these since this one shows up as "merged" for whatever reason

**Comment by adder46** [comment]
It was merged, but I did a hard reset back to your first PR which was okay, and force pushed.

**Comment by adder46** [comment]
I came across [this](https://mypy.readthedocs.io/en/stable/common_issues.html#import-cycles).

**Pull request merged**